src/main/scala/xiangshan/backend/fu/PMP.scala

```diff
@@ -530,12 +551,30 @@ class PMPChecker
 
   val req = io.req.bits
 
-  val res_pmp = pmp_match_res(leaveHitMux, io.req.valid)(req.addr, req.size, io.check_env.pmp, io.check_env.mode, lgMaxSize)
-  val res_pma = pma_match_res(leaveHitMux, io.req.valid)(req.addr, req.size, io.check_env.pma, io.check_env.mode, lgMaxSize)
+  val res_pmp = pmp_match_res(leaveHitMux, io.req.valid)(req.addr(PMPAddrBits-PMPKeyIDBits-1, 0), req.size, io.check_env.pmp, io.check_env.mode, lgMaxSize)
+  val res_pma = pma_match_res(leaveHitMux, io.req.valid)(req.addr(PMPAddrBits-PMPKeyIDBits-1, 0), req.size, io.check_env.pma, io.check_env.mode, lgMaxSize)
```

Comment:
Is this behaviour correct when CVM is disabled?

Reply:
When "HasCVMExtension = Some(true)", the actual physical address bit width change to "PAddrBits-KeyIDBits".
In this config, if CVM is disabled, the KeyID must be "0" and PMP/PMA only need  to check actual physical address bit.
When "HasCVMExtension = Some(false)", KeyIDBits should be set to "0"(I will fix it in file SoC.scals).

Reply:
I suggest adding some comments about what a `KeyID` is and what it does under the abovementioned circumstances.

Reply:
I agree with your suggestion and I will add relevant comments in the next commit.